src/main/java/org/myrobotlab/service/interfaces/EncoderControl.java

```diff
@@ -10,7 +10,7 @@
   public void setController(EncoderController controller);
 
   // TODO: publish encoder data that includes the name/id of the device.
-  public void onEncoderData(EncoderData data);
+  // public void onEncoderData(EncoderData data);

```

Comment:
shouldn't this be renamed to publishEncoderData(EncoderData data)  ?

Reply:
no the EncoderController does the publishing

Reply:
ok.. but ServoController isn't responsible for publishing the data.. that's done by Servo currently.  So this would be a change from that pattern.  Seems like the EncoderControl should be the one publishing..   this makes the Controller more transparent..  

.. as a user..  I would like to listen to servo events from the servo.  

is different than

.. as a user..  I would like to have a servo and I will get events from the controller. 

The first statement seems much simpler and easier to explain to people.

Reply:
I agree ..

I think I went overboard in hating ServoEvents.  I knew a piece was missing (real sensor data).

Now, I'm starting to warm up to the idea to ServoEvents... but it takes EncoderControl, EncoderControllers & EncoderListeners to support them.

I agree EncoderControl should do the publishing - EncoderControllers do the callback into the Control.   And listeners use and handle callbacks from the control.

Yep this has been the pattern, I got too enthusiastic with my hacking away ;)

Will correct ...  again thanks for the catch